api/turing/service/alert_service.go

```diff
@@ -17,17 +18,30 @@ import (
 type Metric string
 
 type AlertService interface {
-	Save(alert models.Alert, authorEmail string) (*models.Alert, error)
+	Save(alert models.Alert, authorEmail string, dashboardURL string) (*models.Alert, error)
 	List(service string) ([]*models.Alert, error)
 	FindByID(id uint) (*models.Alert, error)
-	Update(alert models.Alert, authorEmail string) error
-	Delete(alert models.Alert, authorEmail string) error
+	Update(alert models.Alert, authorEmail string, dashboardURL string) error
+	Delete(alert models.Alert, authorEmail string, dashboardURL string) error
+	GetDashboardURLTemplate() template.Template
 }
 
 type gitlabOpsAlertService struct {
 	db     *gorm.DB       // database client
 	gitlab *gitlab.Client // GitLab client
-	config config.AlertConfig
+	// dashboardURLTemplate is a template for grafana dashboard URL that shows router metric.
+	// The template will be executed with dashboardURLValue.
+	dashboardURLTemplate template.Template
+	config               config.AlertConfig
+}
+
+// dashboardURLValue will be passed in as argument to execute dashboardURLTemplate.
+type DashboardURLValue struct {
+	Environment string // environment name where the router is deployed
+	Cluster     string // Kubernetes cluster name where the router name is deployed
+	Project     string // MLP project name where the router is deployed
+	Router      string // router name for the alert
+	Revision    string // router revision number
```

Comment:
May be `Version` is better to use here instead of `Revision`, since the Turing App UI uses that terminology? It also corresponds to the RouterVersion DB model's Version field.

Reply:
@davidheryanto Not sure you noticed this earlier comment. It's a small thing. I personally think `Version` is better here but I don't have a strong opinion.

Reply:
OK yeah sure version sounds more consistent